### DISCUSSION TITLE
Code Formatting

### DISCUSSION THREAD

**Discussion by lukediamond** [community/community#6659] [Copilot Conversations] [answered]
Hi, I'm wondering if there are plans to allow the format of code suggested to be configured, or detected from editor settings, project style, etc. I've had 4-space tabs inserted for languages set to use 2 spaces, or C++ written with opening braces on their own lines. Assuming the ML component is not configurable enough to mitigate this, would integration with a code formatting tool be a possibility?

**Comment by Maciej93** [comment]
0xA1600E627b3583594d79E12be4f62A92De143322

**Comment by Urethramancer** [comment]
I use external code-formatting tools and format on save. Copilot shouldn't interfere with that - gofmt knows best for Go, Prettier may be the best for other languages.

**Comment by lukediamond** [comment, marked as answer]
My mistake, upon further testing it's clear that although absolute code style is not guaranteed, the engine is very good at keeping _consistency_ with the rest of a codebase/file. K&R, OTBS, and Allman indent styles as well as the different naming conventions are all kept in an intelligent, context-sensitive manner.
